Added GPL licenses

kuniss opened the pull request:
I've added all actual GPL licenses, for this I replaced license aliases string comparison by pattern matching as it is more flexible regarding user specific formatting which may happen under circumstances;
replaced license web references by the origins of license - should be the origin, not opensource.org/licenses;
MIT license recognition corrected too

kuniss committed:
added GPL licenses

for this replaced license aliases string comparison by pattern matching;
replaced license web references by the origins of license

oehme commented on src/test/java/com/github/oehme/sobula/LicenseTest.xtend:
```diff
@@ -0,0 +1,226 @@
+        '''
+        assertTrue(License.LGPL_V2_1.matches(licenseText))
+        
+        assertFalse(License.AGPL_V3_0.matches(licenseText))
```
Can you avoid this repetition by having an `onlyMatchedBy` method?

oehme commented on src/main/java/com/github/oehme/sobula/License.java:
```diff
@@ -1,27 +1,67 @@
-		"http://opensource.org/licenses/MIT"
+		"http://opensource.org/licenses/MIT",
+		Pattern.quote(
+			"Permission is hereby granted, free of charge, to any person obtaining a copy\n" +
```
Why is there the full license text here?

kuniss replied:
As I figured out, the MIT license is not a real license like the EPL or the GPL-like licenses.
It is more a license template. So, you will not find any specific hint in such a particular real license that it was copied from the MIT license template. The only way to recognize it is to check for the whole text (in that sense I should really copy in the while license template text in ...).

oehme commented:
Awesome, just minor comments =)

kuniss committed:
added the full test of MIT license 

except from the header with copyright notice as this is the only part
which differs from user to user;
see https://github.com/oehme/sobula/pull/5 why this was added

kuniss committed:
improved test code applying DRY principle

... as suggested right by Stefan Oehme

kuniss committed:
rewrote test helper method as Xtend extension method 

for better readability

kuniss commented:
BTW, why did you code the License enumeration in Java and not in Xtend? Is where something in Xtend what prevents you from doing so?

oehme commented:
Xtend doesn't allow methods/fields/constructors in Enums.

oehme commented:
Thanks a lot for the contribution =)

oehme commented:
Released as 0.6.6, will be available soon.

kuniss commented:
Thanks, worked we now: https://travis-ci.org/kuniss/ScalaFlow/builds/97300181

kuniss commented:
I meant, ... worked well now.
